docker/Dockerfile.test

```diff
@@ -15,7 +15,7 @@ COPY . .
 
```

Comment:
`contrib/` seems like the wrong location for a Dockerfile considering we have `docker/` so I moved this too.

This file may be entirely removable. Defer to reviewers if that's the case.